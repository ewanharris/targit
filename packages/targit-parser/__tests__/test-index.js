const parseURI = require('../index');

const plainURLCases = [
	[ 'https://www.bitbucket.org/ewanharris/targit' ],
	[ 'https://bitbucket.org/ewanharris/targit' ],
	[ 'https://www.bitbucket.org/ewanharris/targit#abranch' ],
	[ 'https://bitbucket.org/ewanharris/targit#abranch' ],

	[ 'https://www.github.com/ewanharris/targit' ],
	[ 'https://github.com/ewanharris/targit' ],
	[ 'https://www.github.com/ewanharris/targit#abranch' ],
	[ 'https://github.com/ewanharris/targit#abranch' ],

	[ 'https://www.gitlab.com/ewanharris/targit' ],
	[ 'https://gitlab.com/ewanharris/targit' ],
	[ 'https://www.gitlab.com/ewanharris/targit#abranch' ],
	[ 'https://gitlab.com/ewanharris/targit#abranch' ],
];

const userRepoCases = [
	[ { uri: 'ewanharris/targit', host: 'bitbucket' } ],
	[ { uri: 'ewanharris/targit#abranch', host: 'bitbucket' } ],

	[ { uri: 'ewanharris/targit' } ],
	[ { uri: 'ewanharris/targit#abranch' } ],

	[ { uri: 'ewanharris/targit', host: 'gitlab' } ],
	[ { uri: 'ewanharris/targit#abranch', host: 'gitlab' } ],
];

const hostUserRepoCases = [
	[ 'bitbucket:ewanharris/targit' ],
	[ 'bitbucket:ewanharris/targit#abranch' ],

	[ 'github:ewanharris/targit' ],
	[ 'github:ewanharris/targit#abranch' ],

	[ 'gitlab:ewanharris/targit' ],
	[ 'gitlab:ewanharris/targit#abranch' ],
];

const gitHostRepoCases = [
	[ 'git@bitbucket.org:ewanharris/targit.git' ],

	[ 'git@github.com:ewanharris/targit.git' ],

	[ 'git@gitlab.com:ewanharris/targit.git' ],
];

describe('parseURI', () => {

	test('should throw if it can not parse the uri', () => {
		expect(function () {
			parseURI('');
		}).toThrowError(Error, 'Could not parse');
	});

	test.each(plainURLCases)(
		'https url: %s',
		(uri) => {
			const parsed = parseURI(uri);
			expect(parsed).toMatchSnapshot();
		}
	);

	test.each(userRepoCases)(
		'user/repo: %j',
		({ uri, host }) => {
			const parsed = parseURI(uri, host);
			expect(parsed).toMatchSnapshot();
		}
	);

	test.each(hostUserRepoCases)(
		'host:user/repo: %s',
		(uri) => {
			const parsed = parseURI(uri);
			expect(parsed).toMatchSnapshot();
		}
	);

	test.each(gitHostRepoCases)(
		'git@host:user/repo: %s',
		(uri) => {
			const parsed = parseURI(uri);
			expect(parsed).toMatchSnapshot();
		}
	);
});
