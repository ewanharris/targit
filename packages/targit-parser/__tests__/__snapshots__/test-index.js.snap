// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseURI git@host:user/repo: git@bitbucket.org:ewanharris/targit.git 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/master.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI git@host:user/repo: git@github.com:ewanharris/targit.git 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/master.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI git@host:user/repo: git@gitlab.com:ewanharris/targit.git 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=master",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=master",
  },
  "ref": "master",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI host:user/repo: bitbucket:ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/master.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI host:user/repo: bitbucket:ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/abranch.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI host:user/repo: github:ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/master.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI host:user/repo: github:ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/abranch.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI host:user/repo: gitlab:ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=master",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=master",
  },
  "ref": "master",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI host:user/repo: gitlab:ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=abranch",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=abranch",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://bitbucket.org/ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/master.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://bitbucket.org/ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/abranch.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://github.com/ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/master.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://github.com/ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/abranch.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://gitlab.com/ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=master",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=master",
  },
  "ref": "master",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://gitlab.com/ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=abranch",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=abranch",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://www.bitbucket.org/ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/master.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://www.bitbucket.org/ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/abranch.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://www.github.com/ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/master.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://www.github.com/ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/abranch.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://www.gitlab.com/ewanharris/targit 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=master",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=master",
  },
  "ref": "master",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI https url: https://www.gitlab.com/ewanharris/targit#abranch 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=abranch",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=abranch",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI user/repo: {"uri":"ewanharris/targit","host":"bitbucket"} 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/master.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI user/repo: {"uri":"ewanharris/targit","host":"gitlab"} 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=master",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=master",
  },
  "ref": "master",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI user/repo: {"uri":"ewanharris/targit"} 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/master.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/master.zip",
  },
  "ref": "master",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI user/repo: {"uri":"ewanharris/targit#abranch","host":"bitbucket"} 1`] = `
Object {
  "archives": Object {
    "tgz": "https://bitbucket.org/ewanharris/targit/get/abranch.tar.gz",
    "zip": "https://bitbucket.org/ewanharris/targit/get/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "bitbucket",
  "url": "https://bitbucket.org/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI user/repo: {"uri":"ewanharris/targit#abranch","host":"gitlab"} 1`] = `
Object {
  "archives": Object {
    "tgz": "https://gitlab.com/ewanharris/targit/repository/archive.tar.gz?ref=abranch",
    "zip": "https://gitlab.com/ewanharris/targit/repository/archive.zip?ref=abranch",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "gitlab",
  "url": "https://gitlab.com/ewanharris/targit",
  "user": "ewanharris",
}
`;

exports[`parseURI user/repo: {"uri":"ewanharris/targit#abranch"} 1`] = `
Object {
  "archives": Object {
    "tgz": "https://github.com/ewanharris/targit/archive/abranch.tar.gz",
    "zip": "https://github.com/ewanharris/targit/archive/abranch.zip",
  },
  "ref": "abranch",
  "repo": "targit",
  "site": "github",
  "url": "https://github.com/ewanharris/targit",
  "user": "ewanharris",
}
`;
